LaTeX/LaTeX Log.sublime-syntax

```diff
@@ -6,44 +6,81 @@ scope: text.log.latex
 first_line_match: "This is (pdf|pdfe)?TeXk?, Version "
 
 contexts:
+  eol-pop:
+    - match: '$\n?'
+      pop: 1
+
   main:
+    - include: invalids
+    - include: functions
+    - include: types
+    - include: tags
+    - include: comments
+
+    - match: ".*FiXme:"
+      scope: meta.log.latex.fixme
+
+    - match: (?:Overfull|Underfull)
+      scope: keyword.control.hyphenation.latex
+      push:
+        - meta_scope: meta.log.latex.hyphenation
+
+        - match: \[\]\n
+          scope: keyword.control.hyphenation.latex
+          pop: 1
+
+        - match: '[0-9]+\-\-[0-9]+'
+          scope: variable.parameter.hyphenation.latex2
+
+    - include: strings
+
+  invalids:
     - match: ".*Warning:"
       scope: invalid.deprecated
+
     - match: '[^:]*:\d*:.*'
       scope: invalid.deprecated
+
     - match: .*Error|^!.*
       scope: invalid.illegal
+
+  functions:
     - match: .*\.sty
       scope: entity.name.function
+
+  types:
     - match: .*\.cls
       scope: entity.name.type.class
+
+  tags:
     - match: .*\.cfg
       scope: entity.name.tag.configuration
+
     - match: .*\.def
       scope: entity.name.tag.definition
```

Comment:
All of those just seem to address different kinds of file paths. Putting them into a single context would probably be enough.